scripts/init.d-attach-static-vdis

```diff
@@ -32,23 +32,48 @@ clear_stale_state(){
 }
 
 attach_all(){
+	RC=0
 	ALL=$(ls -1 ${STATE_DIR})
 
 	for i in ${ALL}; do
 	    UUID=$(cat ${STATE_DIR}/${i}/vdi-uuid)
 	    logger "Attempting to attach VDI: ${UUID}"
 	    OUTPUT=$(/opt/xensource/bin/static-vdis attach ${UUID} 2>&1)
 	    if [ $? -ne 0 ]; then
+	       RC=1
 	       logger "Attempt to attach VDI: ${UUID} failed -- skipping (Error was: ${OUTPUT})"
 	    fi
 	done
+	return $RC
+}
+
+detach_all(){
+	ALL=$(ls -1 ${STATE_DIR})
+
+	for i in ${ALL}; do
+	    UUID=$(cat ${STATE_DIR}/${i}/vdi-uuid)
+	    logger "Attempting to detach VDI: ${UUID}"
+	    OUTPUT=$(/opt/xensource/bin/static-vdis detach ${UUID} 2>&1)
+	    if [ $? -ne 0 ]; then
+	       logger "Attempt to detach VDI: ${UUID} failed -- skipping (Error was: ${OUTPUT})"
+	    fi
+	done
 }
 
 start() {
 	echo -n $"Attempting to attach all statically-configured VDIs"
 	clear_stale_state
 	attach_all
+	RC=$?
 	echo 
+	return $RC
+}
+
+
+stop() {
+	echo -n $"Attempting to detach all statically-configured VDIs"
```

Comment:
What is `echo $"..."` doing different from `echo "..."`? I'm not familiar with this. It is also used above.